### ISSUE TITLE
Ref gives the wrong element after the first render

### ISSUE THREAD

**Issue by chrisdavies** [developit/preact#87]
I have a component called `AppPane` which has this render method (simplified):

```js
render({children}) {
    return (
      <div ref={el => {
          console.log(el)
          this.rootEl = el
        }}>
        {children}
      </div>
    )
  }
```

The first `console.log` correctly logs the `div` element. But subsequent renders log `AppPane`. I can still access `this.rootEl.base` to get access to the div, so I do have a workaround.

**Comment by developit** [comment]
Hi Chris - to clarify: is the subsequent re-rendering triggered by setState()/forceUpdate()?

**Comment by chrisdavies** [comment]
Hey. Sorry. It's by `setState` in a parent component. Basically, I have state in a parent that is determining which `AppPane` is current.

**Comment by chrisdavies** [comment]
I can try to put together a small example.

**Comment by developit** [comment]
would be useful. Try webpackbin.com if you haven't yet, it's awesome. :+1:

**Comment by chrisdavies** [comment]
I'd never heard of webpackbin... It *is* awesome.

Here's the repro. You can click anywhere on the screen to increment the counter. Open up the browser console and you'll see the issue:

http://www.webpackbin.com/E1WAAp22x

**Comment by developit** [comment]
Fix for this is released in [4.3.0](https://github.com/developit/preact/releases/tag/4.3.0).
